docs/docs/integrations/chat/oci_model_deployment_endpoint.ipynb

```diff
@@ -0,0 +1,414 @@
+    "\n",
+    "This notebooks goes over how to use a chat model hosted on a [OCI Data Science Model Deployment Service](https://docs.oracle.com/en-us/iaas/data-science/using/model-dep-about.htm).\n",
+    "\n",
+    "For authentication, the [oracle-ads](https://accelerated-data-science.readthedocs.io/en/latest/user_guide/cli/authentication.html) library is used to automatically load credentials required for invoking the endpoint.\n",
+    "\n",
+    "## Set up\n",
+    "\n",
+    "You must deploy a chat model on OCI Data Science Model deployment and obtain the following parameters:\n",
+    "\n",
+    "* `endpoint` : The REST endpoint url provided by the endpoint.\n",
+    "\n",
+    "### Deploy model\n",
+    "You can easily deploy, fine-tune, and evaluate foundation models using the [AI Quick Actions](https://docs.oracle.com/en-us/iaas/data-science/using/ai-quick-actions.htm) on OCI Data Science Model deployment. For additional deployment examples, please visit the [Oracle GitHub samples repository](https://github.com/oracle-samples/oci-data-science-ai-samples/blob/main/ai-quick-actions/deploy-with-smc.md).\n",
+    "\n",
+    "### Policies\n",
+    "Make sure to have the required [policies](https://docs.oracle.com/en-us/iaas/data-science/using/model-dep-policies-auth.htm#model_dep_policies_auth__predict-endpoint) to access the OCI Data Science Model Deployment endpoint.\n",
+    "\n",
+    "### Authentication\n",
+    "\n",
+    "You can set authentication through either ads or environment variables. When you are working in OCI Data Science Notebook Session, you can leverage resource principal to access other OCI resources. Check out [here](https://accelerated-data-science.readthedocs.io/en/latest/user_guide/cli/authentication.html) to see more options. \n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Installation"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%pip install -qU langchain-community oracle-ads"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Instantiation"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import ads\n",
+    "from langchain_community.chat_models import ChatOCIModelDeploymentEndpoint\n",
+    "\n",
+    "# Set authentication through ads\n",
+    "# Use resource principal are operating within a\n",
+    "# OCI service that has resource principal based\n",
+    "# authentication configured\n",
+    "ads.set_auth(\"resource_principal\")\n",
+    "\n",
+    "# Create an instance of OCI Model Deployment Endpoint\n",
+    "# Replace the endpoint uri with your own\n",
+    "# Using generic class as entry point, you will be able\n",
+    "# to pass model parameters through model_kwargs during\n",
+    "# instantiation.\n",
+    "chat = ChatOCIModelDeploymentEndpoint(\n",
```

Comment:
Maybe it is better to add a description about when to use `ChatOCIModelDeploymentEndpoint` and when to use `ChatOCIModelDeploymentEndpointVLLM`. The two cells look almost the same :)

Reply:
Added more description in doc, as well as provided `api_key` auth example in the second cell.